examples/medplum-scheduling-demo/src/bots/core/set-availability.ts

```diff
@@ -0,0 +1,115 @@
+import { BotEvent, MedplumClient, resolveId } from '@medplum/core';
+import { Bundle, BundleEntry, Reference, Resource, Schedule, Slot } from '@medplum/fhirtypes';
+
+export interface SetAvailabilityEvent {
+  schedule: Reference<Schedule>;
+  startDate: string;
+  endDate: string;
+  startTime: string;
+  endTime: string;
+  duration: number;
+  daysOfWeek: string[];
+}
+
+export async function handler(medplum: MedplumClient, event: BotEvent<SetAvailabilityEvent>): Promise<Bundle> {
+  const { schedule, startDate, endDate, startTime, endTime, duration, daysOfWeek } = event.input;
+
+  // Basic data validation
+  if (duration <= 0) {
+    throw new Error('Duration must be a positive number');
+  }
+  if (new Date(endDate) < new Date(startDate)) {
+    throw new Error('End date must be after start date');
+  }
+
+  // Update the period of time covered by schedule
+  await medplum.patchResource('Schedule', resolveId(schedule) as string, [
+    {
+      op: 'add',
+      path: '/planningHorizon',
+      value: {
+        start: new Date(`${startDate}T${startTime}.000Z`),
+        end: new Date(`${endDate}T${endTime}.000Z`),
+      },
+    },
+  ]);
```

Comment:
@rahul1 I wonder if using those dates here is a suitable approach, given that the user can trigger the bot multiple times due to the app's design.

Reply:
@pamella  let’’s actually omit the planningHorizon for now. The semantics of that field are slightly different - it’s the horizon for the whole schedule, not just the current block

Reply:
All right, done